enhancements/insights/config-api.md

```diff
@@ -0,0 +1,287 @@
+
+Mitigation: Maybe we should introduce some limit on the number of forced gatherings within 1 hour, for example. 
+
+## Design Details
+
+### On demand data gathering
+
+The on demand data gathering can happen almost immediately when there is no active gathering in the progress, but when there is some in progress 
+then it will be interrupted and the new gathering will be triggered. 
+
+The forced gathering is defined as follows: 
+- force it - e.g using a `ForceGatherinReason` attribute where the user would provide some string as reason (mentioned in the [Proposal](#proposal)). 
+  - pros
+    - [providing a reason (e.g “Refresh Insights”) is simpler for configuration than providing a time
+    - gathering would be triggered almost immediately
+    - there is already similar precedent with `ForceRedeploymentReason` when redeploying static pods
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
+
+### Open Questions
+
+- ~~Can the API define default values? How does it work or how is it implemented?~~ This is done by using `// +kubebuilder:default=`
+- ~~Should the new time of forced gathering (if any defined by user) set a new time for the existing gathering interval? If I force gathering at time A, will the new automatic gathering be in time A + interval?~~ - yes. It makes sense to update the last known time of gathering so that the regular gathering does not happen immediately after the forced gathering. 
+
+### Test Plan
+
+The new config API will require updates of the existing Insights operator testsuite. The testsuite will be extended to cover the new API and respective attributes. 
+Following behavior will need to be tested:
+- `support` secret config takes precedence over the new config API
+- forced gathering works as expected
+- disabling some gatherers works as expected and diasbling all will disable the gathering completely
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API as the `v1alpha1` version and it will be behind `TechPreviewNoUpgrade` feature gate. We would like to
+keep the original way of configuring the Insights operator (that is via `support` secret), because it is problematic to move every existing configuration option to the new configuration API and the existing `support` secret serves basically as a configuration API. 
+
+#### Dev Preview -> Tech Preview
+
+The original way of configuring the Insights is still available. If there is a conflict, in other words, if there is some configuration option defined in the `support` secret as well as in the new configuration API then the value from the `support` secret takes precedence. 
+
+#### Tech Preview -> GA
+
+The `TechPreviewNoUpgrade` feature gate requirement is removed. The behavior defined above in the [Dev Preview -> Tech Preview](#dev-preview---tech-preview) section
```

Comment:
How will you know when you're ready for GA?

Reply:
Duplicate. Resolving.